### PR TITLE
Nix

### PR DESCRIPTION
**Pull request by robwhitaker**
Redid the dependency management with Nix. No global dependencies should need to be installed other than Nix to build the project now. Updated the README to reflect the changes.

### CODE REVIEW

**Comment by larouxn** [comment]
Is this ready to be reviewed/checked? Should I do so on Linux (staging)?

**Comment by robwhitaker** [comment]
Yep, go ahead. Trying to go for a clean build on staging would be good.

**Comment by larouxn** [comment]
Alright, actually, I think I'll spin up a local Lubuntu VM so I don't break staging unnecessarily.

**Comment by larouxn** [comment]
Looks like you're requiring `Gemfile.lock` but it doesn't exist yet.

I forgot to post my PR that adds it.https://github.com/robwhitaker/mmp-website/pull/140

Once that's merged we can run, `git rebase origin/master` on your branch/PR.

File: README.md [hunk 1]
```diff
@@ -1,35 +1,58 @@
...
 - `ruby app.rb`
 - Visit `localhost:4567`/`localhost:4567/editor` in your browser
+
+### A Note On `nix-shell`
+
+You don't technically have to be in a Nix shell session to run the above commands, but it will save you some key strokes. You could also run most commands with `nix-shell shell.nix --pure --run "COMMAND"`. For example, to start the server from outside the Nix shell:
+
+```
+$ nix-shell shell.nix --pure --run "ruby app.rb"
+```
```
Comment (larouxn) [inline README.md:58]: Wasn't entirely clear to me I needed to run this as opposed to just `ruby app.rb`, though it does make sense in retrospect.

**Review by larouxn — APPROVED** [review]
LGTM 👍 ⛵️ after https://github.com/robwhitaker/mmp-website/pull/140.

**Comment by larouxn** [comment]
For posterity, I think diffs of this size are pure insanity.
screen shot 2018-08-07 at 13 47 31

**Pull request merged**